File: src/homemade/game/fieldstructure/LinkCode.java
package homemade.game.fieldstructure;

/**
 * This class represents link as a part of the field structure.
 */
public final class LinkCode {

    private CellCode higher, lower;
    private Direction direction;

    private int code;

    LinkCode(Direction direction, CellCode lower, CellCode higher, int code) {
        this.direction = direction;

        this.lower = lower;
        this.higher = higher;

        this.code = code;
    }

    @Override
    public int hashCode() {
        return code;
    }

    public CellCode getLower() {
        return lower;
    }

    public CellCode getHigher() {
        return higher;
    }

    public Direction getLowerToHigherDirection() {
        return direction;
    }
}

File: field-structure/src/homemade/game/fieldstructure/LinkCode.kt
package homemade.game.fieldstructure

class LinkCode internal constructor(val lowerToHigherDirection: Direction,
                                    @JvmField val lower: CellCode,
                                    @JvmField val higher: CellCode,
                                    private val code: Int) {
    override fun hashCode(): Int {
        return code
    }

    override fun equals(other: Any?): Boolean {
        return this === other
    }
}
